### PR TITLE
638 notify admin promotion on user creation

### PR DESCRIPTION
**Pull request by chobeat**
- fixed createuser render test
- fixed some create_user tests
- fixed create_user tests
- added flash notification for first user autopromotion to admin

### CODE REVIEW

File: lib/views/create_user/create_user_controller.ex [hunk 1]
```diff
@@ -26,11 +26,14 @@ defmodule Bonfire.UI.Me.CreateUserController do
            unindexable: not empty?(Map.get(params, "unindexable")),
            request_before_follow: not is_nil(Map.get(params, "request_before_follow"))
          ) do
-      {:ok, %{id: id, profile: %{name: name}} = _user} ->
-        greet(conn, params, id, name)
+      {:ok, %{id: id, profile: %{name: name}} = user} ->
+        is_admin = Accounts.is_admin?(user)
```
Comment (mayel) [inline lib/views/create_user/create_user_controller.ex]: can be simplied as `greet(conn, params, id, name, Accounts.is_admin?(user))`

File: lib/views/create_user/create_user_controller.ex [hunk 2]
```diff
@@ -55,11 +58,22 @@ defmodule Bonfire.UI.Me.CreateUserController do
     end
   end
 
-  defp greet(conn, params, id, name) do
+  defp build_greet_message(name, is_admin) do
```
Comment (mayel) [inline lib/views/create_user/create_user_controller.ex]: can be simplified as (note also the `l` function for localisation)
```
  defp build_greet_message(name, is_admin) do
    flash_message = l("Hey %{name}, nice to meet you!", name: name)

      if is_admin do
        flash_message <> " " <> l("You have been promoted to admin!")
      else
        flash_message
      end
  end
```

or 
```
  defp build_greet_message(name, true) do
    build_greet_message(name, nil) <> " " <> l("You have been promoted to admin!")
  end
  defp build_greet_message(name, _) do
    l("Hey %{name}, nice to meet you!", name: name) 
  end

**Review by mayel — DISMISSED** [review]
awesome, thanks for fixing so many tests!

File: lib/views/create_user/create_user_controller.ex (hunk 2 above)
Comment (chobeat) [reply lib/views/create_user/create_user_controller.ex]: That` if ` felt smelly but I wasn't sure what was the idiomatic way of doing it. I will update it

**Comment by mayel** [comment]
thanks again! closes https://github.com/bonfire-networks/bonfire-app/issues/638

**Pull request merged**